### ISSUE TITLE
Create user “tss” on “make install”

### ISSUE THREAD

**Issue by dilyanpalauzov** [tpm2-software/tpm2-abrmd#613]

**Comment by diabonas** [comment]
A couple of concerns regarding this approach:

- If this is done unconditionally, installation will require root privileges, so you can't simply install to a user-accessible directory, e.g. for packaging purposes.
- Speaking of packaging, creating the necessary users and groups is something the package manager of your favourite distribution should take care of (which should IMHO be the preferred way for end users to install software, as opposed to manually dropping files into `/usr/local`).
- I think tpm2-tss would be the better place for the `tss` user because the [udev rules](https://github.com/tpm2-software/tpm2-tss/blob/master/dist/tpm-udev.rules) that give the user access to the TPM are packaged there. It seems more logical to keep the user and rules creation together, and since tpm2-abrmd depends on tpm2-tss, the user will be available here as well.

Instead of manually creating the user in tpm2-abrmd, I would suggest adding a [sysusers.d file](https://www.freedesktop.org/software/systemd/man/sysusers.d.html) to tpm2-tss, enabling automatic creation of the user if necessary using systemd. This is the approach that was chosen in [Arch Linux](https://git.archlinux.org/svntogit/community.git/tree/trunk/PKGBUILD?h=packages/tpm2-tss&id=d2f10e592fe8c3f75e4cde46de2dc6dec2c25014#n32).

**Comment by dilyanpalauzov** [comment]
If the “tss” user is not installed on “make install”, but the file “tpm2-abrmd.service” is, the latter should not contain “User=tss”.

**Comment by diabonas** [comment]
The reason for `User=tss` is that the udev rule installed by *tpm2-tss* gives access to the TPM to this user, so it should be the job of tpm2-tss to guarantee that this user actually exists. Removing the line would work in theory, but this is suboptimal as the service is run as root then.

**Comment by flihp** [comment]
Thanks for getting this conversation started. Since tabrmd is selecting defaults based on defaults from tpm2-tss I've created https://github.com/tpm2-software/tpm2-tss/issues/1393 to get the discussion started over there. I'm closing this thread down to keep the conversation on the tpm2-tss tracker, but I'll create a new issue here to ensure any defaults changed upstream are accounted for in tabrmd.
